ledger/block/src/helpers/target.rs

```diff
@@ -35,20 +58,83 @@ pub const fn block_reward(total_supply: u64, block_time: u16, coinbase_reward: u
     block_reward + (coinbase_reward / 3) + transaction_fees
 }
 
+/// Calculate the V2 block reward, given the total supply, block interval, coinbase reward, and transaction fees.
+///     R_staking = floor((0.05 * S) * clamp(I, MIN_BI, MAX_BI) / S_Y) + CR / 3 + TX_F.
+///     S = Total supply.
+///     I = Seconds elapsed since last block.
+///     S_Y = Seconds in a year (31536000).
+///     CR = Coinbase reward.
+///     TX_F = Transaction fees.
+///     MIN_BI = Minimum block interval.
+///     MAX_BI = Maximum block interval.
+pub fn block_reward_v2(
+    total_supply: u64,
+    time_since_last_block: i64,
+    coinbase_reward: u64,
+    transaction_fees: u64,
+) -> u64 {
+    // Calculate the number of seconds in a year.
+    const SECONDS_IN_A_YEAR: u64 = 60 * 60 * 24 * 365;
```

Comment:
Defined twice, can be defined at the top of the file.

Reply:
Resolved with [9098995](https://github.com/AleoNet/snarkVM/pull/2569/commits/9098995c53f6d01f0dc2171a6c51d72c8d1b5de7)